Add docs for `HTMLScriptElement.async` and `HTMLScriptElement.defer` (#33286)
mdn/content: files/en-us/web/html/element/script/index.md

    For [module scripts](/en-US/docs/Web/JavaScript/Guide/Modules), if the `async` attribute is present then the scripts and all their dependencies will be fetched in parallel to parsing and evaluated as soon as they are available.

    > **Warning:** This attribute must not be used if the `src` attribute is absent (i.e. for inline scripts) for classic scripts, in this case it would have no effect.

    This attribute allows the elimination of **parser-blocking JavaScript** where the browser would have to load and evaluate scripts before continuing to parse. `defer` has a similar effect in this case.

>     The `async` attribute takes higher priority than the `defer` attribute, which means that, if the two attributes both specified, the `async` attribute will take effect.

Author: ```suggestion
    If the attribute is specified with the `defer` attribute, the element will act as if only the `async` attribute is specified.
```

    This attribute allows the elimination of **parser-blocking JavaScript** where the browser would have to load and evaluate scripts before continuing to parse. `async` has a similar effect in this case.

>     The `defer` attribute takes lower priority than the `async` attribute, which means that, if the two attributes both specified, the `defer` attribute will not take effect.

Author: ```suggestion
    If the attribute is specified with the `async` attribute, the element will act as if only the `async` attribute is specified.
```
